internal/storage/auth/keycloak/token_test.go

```diff
@@ -0,0 +1,118 @@
+package keycloak
+
+import (
+	"context"
+	"testing"
+
+	"github.com/green-ecolution/green-ecolution-backend/internal/entities"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestKeyCloakRepo_RetrospectToken(t *testing.T) {
+	t.Run("should return valid token", func(t *testing.T) {
+		// given
+		ctx := context.Background()
+		cfg := suite.IdentityConfig(t, ctx)
+		k := NewKeycloakRepository(cfg)
+
+		validToken, err := loginRestAPIClient(ctx, cfg.KeyCloak.BaseURL, cfg.KeyCloak.ClientID, cfg.KeyCloak.ClientSecret, cfg.KeyCloak.Realm)
+		assert.NoError(t, err)
+
+		// when
+		got, err := k.RetrospectToken(ctx, validToken.AccessToken)
+
+		// then
+		assert.NoError(t, err)
+		assert.NotNil(t, got)
+		assert.NotNil(t, got.Active)
+		assert.True(t, *got.Active)
+	})
+
+	t.Run("should return not active token when token is invalid", func(t *testing.T) {
+		// given
+		ctx := context.Background()
+		cfg := suite.IdentityConfig(t, ctx)
+		k := NewKeycloakRepository(cfg)
+
+		// when
+		got, err := k.RetrospectToken(ctx, "invalid-token")
+
+		// then
+		assert.NotNil(t, got)
+		assert.NotNil(t, got.Active)
+		assert.False(t, *got.Active)
+		assert.NoError(t, err)
+	})
+
+	t.Run("should return not active token when token is empty", func(t *testing.T) {
+		// given
+		ctx := context.Background()
+		cfg := suite.IdentityConfig(t, ctx)
+		k := NewKeycloakRepository(cfg)
+
+		// when
+		got, err := k.RetrospectToken(ctx, "")
+
+		// then
+		assert.NotNil(t, got)
+		assert.NotNil(t, got.Active)
+		assert.False(t, *got.Active)
+		assert.NoError(t, err)
+	})
+}
+
+func TestKeyCloakRepo_GetAccessTokenFromClientCode(t *testing.T) {
+	t.Skip("don't know how to test this yet, need to recreate the login flow. Will come back to this later")
```

Comment:
Let's add a TODO and an issue? Otherwise we'll forget it

Reply:
#225